core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -97,6 +97,49 @@ package object config {
     .toSequence
     .createWithDefault(Nil)
 
+  // Blacklist confs
+  private[spark] val BLACKLIST_ENABLED =
+    ConfigBuilder("spark.blacklist.enabled")
+    .booleanConf
+    .createOptional
+
+  private[spark] val MAX_TASK_ATTEMPTS_PER_NODE =
+    ConfigBuilder("spark.blacklist.maxTaskAttemptsPerNode")
+    .intConf
+    .createWithDefault(2)
+
+  private[spark] val MAX_FAILURES_PER_EXEC =
+    ConfigBuilder("spark.blacklist.maxFailedTasksPerExecutor")
+    .intConf
+    .createWithDefault(2)
+
+  private[spark] val MAX_FAILURES_PER_EXEC_STAGE =
+    ConfigBuilder("spark.blacklist.maxFailedTasksPerExecutorStage")
```

Comment:
Actually what if these were

spark.blacklist.executor.maxFailedTasks
spark.blacklist.executor.stage.maxFailedTasks
spark.blacklist.node.maxFailedExecutors
spark.blacklist.node.stage.maxFailedExecutors

Not sure whether this is actually better, just another idea